packages/app/app/components/ListeningHistoryView/PieChartTopArtists.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import {
+  PieChart,
+  Pie,
+  Tooltip,
+  Legend,
+  Cell,
+  Text,
+  ResponsiveContainer
+} from 'recharts';
+
+const colors = [
+  '#FF6633', '#FFB399', '#FF33FF', '#00B3E6', '#E6B333', '#3366E6',
+  '#999966', '#99FF99', '#B34D4D', '#80B300', '#809900', '#E6B3B3',
+  '#6680B3', '#66991A', '#FF99E6', '#CCFF1A', '#FF1A66', '#E6331A',
+  '#33FFCC', '#66994D', '#B366CC', '#4D8000', '#B33300', '#CC80CC',
+  '#66664D', '#991AFF', '#E666FF', '#4DB3FF', '#1AB399', '#E666B3',
+  '#33991A', '#CC9999', '#B3B31A', '#00E680', '#4D8066', '#809980',
+  '#E6FF80', '#1AFF33', '#999933', '#FF3380', '#CCCC00', '#66E64D',
+  '#4D80CC', '#9900B3', '#E64D66', '#4DB380', '#FF4D4D', '#99E6E6',
+  '#6666FF'
+];
+
+// Function to get the top N items from the data based on a given key
+const getTopN = (data: any[], key: string, n: number): any[] => {
+  return data?.sort((a, b) => b[key] - a[key]).slice(0, n);
+};
+
+export type PieChartTopArtists = {
+    data: any[];
+}
+
+const timeframes = ['Last 7 days', 'Last 30 days', 'Last 90 days', 'Last 180 days', 'Last 365 days', 'All time'];
```

Comment:
Nuclear is translated into more than 30 languages. So we should never use hardcoded strings like these, and instead use `react-i18next`. Try looking for the `useTranslation` hook to see how it's used.

If you decide to move this component to the `ui` package, you can simply feed it strings it needs to render its interface. For an example of how it can be done, you can look for `TrackPopupStrings`. The idea is that the component in `ui` takes a list of texts it's going to render, and in the `app` package, when we want to use that component, we look up the translated texts using the `useTranslation` hook. If you're unsure how to do this, let me know.

Reply:
And the same comment applies to other instances of raw text, like `Timeframe`, etc.